networking/metallb/metallb-configure-services.adoc

```diff
@@ -0,0 +1,162 @@
+MetalLB assigns an an IP address from these address pools.
+
+To accept any IP address from any pool that is configured for automatic assignment, no special annotation or configuration is required.
+
+.Example service YAML for accepting any IP address
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <service_name>
+spec:
+  selector:
+    <label_key>: <label_value>
+  ports:
+    - port: 8080
+      targetPort: 8080
+      protocol: TCP
+  type: LoadBalancer
+----
+
+[id="share-specific-ip-address_{context}"]
+== Share a specific IP address
+
+By default, services do not share IP addresses.
+However, if you need to colocate services on a single IP address, you can enable selective IP sharing by adding the `metallb.universe.tf/allow-shared-ip` annotation to the services.
+
+[source,yaml]
+----
+apiVersion: v1
+  kind: Service
+  metadata:
+    name: service-http
+    annotations:
+      metallb.universe.tf/address-pool: doc-example
+      metallb.universe.tf/allow-shared-ip: "web-server-svc"  <1>
+  spec:
+    ports:
+    - name: http
+      port: 80  <2>
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      <label_key>: <label_value>  <3>
+    type: LoadBalancer
+    loadBalancerIP: 172.31.249.7  <4>
+---
+apiVersion: v1
+  kind: Service
+  metadata:
+    name: service-https
+    annotations:
+      metallb.universe.tf/address-pool: doc-example
+      metallb.universe.tf/allow-shared-ip: "web-server-svc"  <1>
+  spec:
+    ports:
+    - name: https
+      port: 443  <2>
+      protocol: TCP
+      targetPort: 8080
+    selector:
+      <label_key>: <label_value>  <3>
+    type: LoadBalancer
+    loadBalancerIP: 172.31.249.7  <4>
+----
+<1> Specify the same value for the `metallb.universe.tf/allow-shared-ip` annotation. This value is referred to as the _sharing key_.
+<2> Specify different port numbers for the services.
+<3> Specify identical pod selectors if you must specify `externalTrafficPolicy: local` so the services send traffic to the same set of pods. If you use the `cluster` external traffic policy, then the pod selectors do not need to be identical.
+<4> Optional: If you specify the three preceding items, MetalLB might colocate the services on the same IP address. To ensure that services share an IP address, specify the IP address to share.
+
+By default, Kubernetes does not allow multiprotocol load balancer services.
+This limitation would normally make it impossible to run a service like DNS that needs to listen on both TCP and UDP.
+To work around this limitation of Kubernetes with MetalLB, create two services:
+
+* For one service, specify TCP and for the second service, specify UDP.
+
+* In both services, specify the same pod selector.
+
+* Specify the same sharing key and `spec.loadBalancerIP` value to colocate the TCP and UDP services on the same IP address.
+
+// Configuring a service with MetalLB
+include::modules/nw-metallb-configure-svc.adoc[leveloffset=+1]
```

Comment:
Add service YAML examples to obtain ipv6 external IP address for the service from configured ipv6 address pool(s).
a) If service is configured on dual stack cluster is 
spec:
  ipFamilyPolicy: RequireDualStack
  ipFamilies:
  - IPv6
  - IPv4

b) This is for configuring service on IPv6 single stack
spec:
  ipFamilies:
  - IPv6

Reply:
@mikemckiernan, related content: `modules/virt-creating-a-service-from-a-virtual-machine.adoc`

Reply:
Federico indicated that MetalLB assigns an IP address based on the IP address family of the cluster IP.